parameter_delegate.cpp

```diff
@@ -161,38 +156,48 @@ namespace dealii
 
               return dirname_editor;
             }
-          else if (rx_integer.indexIn (pattern_description) != -1)		// if the tpye is "Integer"
+          else if (rx_integer.indexIn (pattern_description) != -1)		// if the type is "Integer"
             {
-              QSpinBox * spin_box = new QSpinBox(parent);			// choose a spin box
-
-              const int min_int_value = std::numeric_limits<int>::min();
-              const int max_int_value = std::numeric_limits<int>::max();
+              const QStringList default_pattern = pattern_description.split(" ").filter("...");
+              const QStringList default_values = default_pattern[0].split("...");
 
-              spin_box->setMaximum(max_int_value);				// set max and min from the limits.h class
-              spin_box->setMinimum(min_int_value);
-              spin_box->setSingleStep(int_steps);				// and every klick is a SingleStep
+              QLineEdit * line_edit = new QLineEdit(parent);
+              line_edit->setValidator(new QIntValidator(default_values[0].toInt(), default_values[1].toInt(), line_edit));
```

Comment:
No, you are actively prevented from typing a number that does not fit the range (while typing). Some errors can not be prevented that way though (e.g. for a range of 10 - 20 you have to type a 1 first, which does not fall in the range), however then you are not allowed to press enter until the number is within that range. Pretty neat. I noticed though that you can click another field, even if you entered an unfinished, invalid number, and then the content of the field is invalid.